File: user-management-api/src/main/java/com/usermanagement/demo/controllers/UserController.java
package com.usermanagement.demo.controllers;


import com.usermanagement.demo.entities.User;
import com.usermanagement.demo.services.UserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/users")
@CrossOrigin("*")
public class UserController {

    @Autowired
    private UserService userService;

    @GetMapping
    public List<User> getAllUser(){
        return userService.getAllUser();
    }

    @GetMapping("/{id}")
    public User getUserById(@PathVariable("id") long id){
        return userService.getUserById(id);
    }

    @PostMapping
    public String saveUser(@RequestBody User user){
        userService.addUser(user);
        return "Success";
    }

    @PutMapping("/{id}")
    public String updateuser(@PathVariable("id") long id, @RequestBody User user){
        user.setId(id);
        userService.updateUser(user);
        return "Success";
    }

    @DeleteMapping("/{id}")
    public String deleteUser(@PathVariable("id") long id){
        userService.deleteUserById(id);
        return "Success";
    }

}

File: express-product-management-api/app/controllers/user.controller.js
import userService from "../services/user.services.js";

const authUser = async (req, res) => {
  const token = await userService.authUser(req.body);
  if (!token) {
    res.send({ msg: "invalid credentials" });
  } else {
    res.send({ token: token });
  }
};

const getAllUser = async (req, res) => {
  const users = await userService.getAllUser();
  res.send({ users });
};

const getUserById = async (req, res) => {
  const { id } = req.params;
  const user = await userService.getUserById(id);
  res.send({ user });
};

const addUser = async (req, res) => {
  const user = await userService.addUser(req.body);
  res.send({ user });
};

const updateUser = async (req, res) => {
  const { id } = req.params;
  const user = await userService.updateUser(id, req.body);
  res.send({ user });
};

const deleteUserById = async (req, res) => {
  const { id } = req.params;
  const user = await userService.deleteUserById(id);
  res.send({ user });
};

export {
  getAllUser,
  getUserById,
  addUser,
  updateUser,
  deleteUserById,
  authUser,
};
